src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -251,10 +390,25 @@ internal ImageLoadingEstimator(IHostEnvironment env, string imageFolder, params
         {
         }
 
-        internal ImageLoadingEstimator(IHostEnvironment env, ImageLoadingTransformer transformer)
+        /// <summary>
+        /// Load images in memory.
+        /// </summary>
+        /// <param name="env">The host environment.</param>
+        /// <param name="imageFolder">Folder where to look for images.</param>
+        /// <param name="type">Image type - VectorDataView type or ImageDataViewType. Defaults to ImageDataViewType if not specified or null.</param>
+        /// <param name="columns">Names of input and output columns.</param>
+        internal ImageLoadingEstimator(IHostEnvironment env, string imageFolder, bool type = true, params (string outputColumnName, string inputColumnName)[] columns)
+            : this(env, new ImageLoadingTransformer(env, imageFolder, type, columns), type)
+        {
+        }
+
+        internal ImageLoadingEstimator(IHostEnvironment env, ImageLoadingTransformer transformer, bool type = true)
             : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ImageLoadingEstimator)), transformer)
         {
-            _imageType = new ImageDataViewType();
+            if (type)
+                _type = new ImageDataViewType();
+            else
+                _type = new VectorDataViewType(NumberDataViewType.Byte);
```

Comment:
>_type [](start = 16, length = 5)

Instantiating `_type` this way for the byte array case is incorrect, since it instantiates a variable length vector. It is also not used in this case (see line 432).
I suggest removing the field and using `new ImageDataViewType()` in line 430 instead. (or having a static field of type `ImageDataViewType` if you don't want to allocate a new one on each call, although `GetOutputSchema` is typically not called many times).

Reply:
Changed.